datanode/api/trading_data_v2.go

```diff
@@ -267,11 +264,49 @@ func (t *TradingDataServiceV2) ListLedgerEntries(ctx context.Context, req *v2.Li
 		return nil, formatE(err)
 	}
 
+	leEedges := make([]*v2.LedgerEntriesEdge, 0, len(edges))
+	for _, agg := range edges {
+		if agg == nil || agg.Node == nil {
```

Comment:
this looks, to me at least, like we're accounting for a mistake elsewhere. If the data-set is empty, I'd expect `edges` to be empty, rather than `[]T{nil, T{Node: nil}}` or something